### DISCUSSION TITLE
Is it possible to define some condition names for conditional compilation directives ?

### DISCUSSION THREAD

**Discussion by ThomasBarberot** [codefori/vscode-ibmi#1028] [Q&A]
Hi.
I usually use conditional compilation directive statement to include or exclude sections of source code when I use the /include directive.
I use it with the directive /define or /undefine.

In the source to be included, you check if a condition is defined (if not, the section will not be included)
```
/if defined(API_H_OBJ)
// Avoid multiple inclusions
/if not defined(API_H_OBJ_ALREADY_DONE)
/define API_H_OBJ_ALREADY_DONE

// Change Object Description (QLICOBJD) API
dcl-pr QLICOBJD extpgm('QLICOBJD');
 ...
end-pr;
...
/endif // /if not defined(API_H_OBJ_ALREADY_DONE)
/endif // /if defined(API_H_OBJ)`
```
and in the source in which you want to include only a section of the first source, you define a condition before including the source
```
/define API_H_OBJ
/include api_h
```

In VSC, in the structure view, on the source to included, the analyse of the struture is stopped when the first check of condition is reach, and in the source where the inclusion is done, definitions of declarations included are not found.

In RDi, you can declare some condition to let the structure to correctly analyse both sources.

In VSC, is there the same facilities to declare conditions ?

Thanks.

**Comment by worksofliam** [comment]
There is not today, but this is an excellent feature request. I will look into how we can do this easily. Thanks!
